src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -1,227 +1,280 @@
+            bidType = getBidType(bid, bidRequest.getImp());
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
         }
-        final List<BidderError> errors = new ArrayList<>();
-        final List<BidderBid> bidderBids = extractBids(bidRequest, consumableResponse.getDecisions());
-        return Result.of(bidderBids, errors);
-    }
-
-    private static List<BidderBid> extractBids(BidRequest bidRequest,
-                                               Map<String, ConsumableDecision> impIdToDecisions) {
-        final List<BidderBid> bidderBids = new ArrayList<>();
-        for (Map.Entry<String, ConsumableDecision> entry : impIdToDecisions.entrySet()) {
-            final ConsumableDecision decision = entry.getValue();
-
-            if (decision != null) {
-                final ConsumablePricing pricing = decision.getPricing();
-                if (pricing != null && pricing.getClearPrice() != null) {
-                    final String impId = entry.getKey();
-
-                    final Bid bid = Bid.builder()
-                            .id(bidRequest.getId())
-                            .impid(impId)
-                            .price(BigDecimal.valueOf(pricing.getClearPrice()))
-                            .adm(CollectionUtils.isNotEmpty(decision.getContents())
-                                    ? decision.getContents().get(0).getBody() : "")
-                            .w(decision.getWidth())
-                            .h(decision.getHeight())
-                            .crid(String.valueOf(decision.getAdId()))
-                            .exp(30)
-                            .build();
-                    // Consumable units are always HTML, never VAST.
-                    // From Prebid's point of view, this means that Consumable units
-                    // are always "banners".
-                    bidderBids.add(BidderBid.of(bid, BidType.banner, null));
+
+        final Bid updatedBid = switch (bidType) {
+            case video -> updateBidWithVideoAttributes(bid);
+            default -> bid;
+        };
+
+        return BidderBid.of(updatedBid, bidType, bidResponse.getCur());
+    }
+
+    private Bid updateBidWithVideoAttributes(Bid bid) {
+        final ObjectNode bidExt = bid.getExt();
+        final ExtBidPrebid extPrebid = bidExt != null ? parseBidExt(bidExt) : ExtBidPrebid.builder()
+                .video(ExtBidPrebidVideo.of(null, null)).build();
+        final ExtBidPrebidVideo extVideo = extPrebid != null ? extPrebid.getVideo() : null;
+        final Bid updatedBid;
+        if (extVideo != null) {
+            final Bid.BidBuilder bidBuilder = bid.toBuilder();
+            bidBuilder.ext(resolveBidExt(bid.getDur()));
+            updatedBid = bidBuilder.build();
+        } else {
+            updatedBid = bid;
+        }
+        return updatedBid;
+    }
```

Comment:
imho an Optional chain will fit here, since there are a lot of not-null checks that could be simplified with the Optional.ofNullable(...)